x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -683,7 +683,11 @@ export function resultsServiceProvider(mlClient: MlClient, client?: IScopedClust
 
     const rangeFilter = {
       range: {
-        [timefield]: { gte: start, lte: end },
+        [timefield]: {
+          gte: start,
+          lte: end,
+          format: 'epoch_millis',
```

Comment:
This is the right format, _providing_ `start` and `end` are always in epoch millis. It looks like they are certainly numbers based on: https://github.com/elastic/kibana/blob/b30f8a6ba5dbf26b540edf940d2d70b9daf56063/x-pack/plugins/ml/server/routes/schemas/results_service_schema.ts#L112-L113

But could these numbers ever represent anything that's not epoch millis?

Reply:
Confirmed that start and end in the UI code are definitely always milliseconds for the datafeed. 👍